### PR TITLE
Fix 2 test cases and modify the behaviour of genRandomBits

### PR DESCRIPTION
**Pull request by ghost**
1- Fix TestSecuredGenRandomBits and TestGenCoPrime to avoid 0 bit.
2- Modify the behaviour of genRandomBits when param 'bits' is set to 1.

### CODE REVIEW

**Comment by ghost** [comment]
Other several tests for obvious methods like min, max, etc. are also added here.

File: BigInteger/BigInteger.cs [hunk 1]
```diff
@@ -1625,25 +1625,36 @@ public BigInteger gcd(BigInteger bi)
     /// <param name="rand"></param>
     public void genRandomBits(int bits, Random rand)
     {
+
```
Comment (bazzilic) [inline BigInteger/BigInteger.cs]: Unnecessary empty line here

File: BigInteger/BigInteger.cs [hunk 2]
```diff
@@ -1625,25 +1625,36 @@ public BigInteger gcd(BigInteger bi)
...
-            uint mask = (uint)(0x01 << (remBits - 1));
-            data[dwords - 1] |= mask;
+            uint mask;
+
+            if (bits != 1)
+            {
+                mask = (uint)(0x01 << (remBits - 1));
+                data[dwords - 1] |= mask;
+            }
+
```
Comment (bazzilic) [inline BigInteger/BigInteger.cs:1657]: Unnecessary empty line here

File: BigInteger/BigInteger.cs [hunk 3]
```diff
@@ -1671,21 +1682,29 @@ public void genRandomBits(int bits, RNGCryptoServiceProvider rng)
...
```
Comment (bazzilic) [inline BigInteger/BigInteger.cs:1705]: Unnecessary empty line here

File: BigIntegerTests/Methods.cs [hunk 4]
```diff
@@ -143,78 +143,316 @@ public void TestJacobi()
         [TestMethod]
         public void TestSecuredGenRandomBits()
         {
-            { // Test < 32 bits
-                var bi = new BigInteger();
-                var rng = new RNGCryptoServiceProvider();
-                var rand = new Random();
+            for (int i = 0; i < 9999; i++)
```
Comment (bazzilic) [inline BigIntegerTests/Methods.cs]: 9999 might be too much for actual tests

File: BigIntegerTests/Methods.cs [hunk 5]
```diff
@@ -143,78 +143,316 @@ public void TestJacobi()
...
+
+            bi2 = new BigInteger("-8979479943434898397", 10);
+            Assert.AreEqual(bi2, bi1.max(bi2));
+
+            bi2 = new BigInteger("-9852375989470234802399402398409238049238094723094709234702387498237498623948623984623984623864237642376482376492386479238749823749237498237498237", 10);
+            Assert.AreEqual(bi1, bi2.max(bi1));
+
+        }
+
+        [TestMethod()]
```
Comment (bazzilic) [inline BigIntegerTests/Methods.cs]: Inconsistent attributes, some with brackets, some without

**Review by bazzilic — CHANGES_REQUESTED** [review]
Looks good overall, just small fixes before merge

File: BigInteger/BigInteger.cs [hunk 6]
```diff
@@ -1671,21 +1681,27 @@ public void genRandomBits(int bits, RNGCryptoServiceProvider rng)
...
 
         for (int i = 0; i < dwords; i++)
-        {
-            rng.GetBytes(randomBytes);
-            data[i] = BitConverter.ToUInt32(randomBytes, 0);
-        }
+            data[i] = BitConverter.ToUInt32(randomBytes, i * 4);
+
+        for (int i = dwords; i < maxLength; i++)
+            data[i] = 0;
```
Comment (bazzilic) [inline BigInteger/BigInteger.cs:1694]: Not necessary to zero the rest since `dataLength` tells us which byte is the last.

**Pull request merged**